Header header
float64[] state_of_charge